src/asciidoctor-extensions-api.js

```diff
@@ -137,3 +148,18 @@ BlockProcessor.$$proto.named = function (name) {
 BlockProcessor.$$proto.onContext = function (context) {
   return this.$on_context(context);
 };
+
+/**
+ * @namespace
+ * @module Extensions/IncludeProcessor
+ */
+var IncludeProcessor = Extensions.IncludeProcessor;
+
+/**
+ * @memberof Extensions/IncludeProcessor
+ */
+IncludeProcessor.$$proto.handles = function (block) {
+  // FIXME: Dirty monkey patch !
+  IncludeProcessor.$$proto['$handles?'] = block;
```

Comment:
I filed an issue, sent a PR, and verified that it fixes your hack. I've been meaning to fix this for a long time. I'm glad that :sun_with_face: has come.

https://github.com/asciidoctor/asciidoctor/pull/2120